examples/model_selection/plot_cv_indices.py

```diff
@@ -0,0 +1,147 @@
+# Define a function to visualize cross-validation behavior
+# --------------------------------------------------------
+#
+# We'll define a function that lets us visualize the behavior of each
+# cross-validation object. We'll perform 5 splits of the data. On each
+# split, we'll visualize the indices chosen for the training set
+# (in blue) and the test set (in red).
+
+
+def plot_cv_indices(cv, X, y, group, ax, n_splits, lw=10):
+    """Create a sample plot for indices of a cross-validation object."""
+
+    # Generate the training/testing visualizations for each CV split
+    for ii, (tr, tt) in enumerate(cv.split(X=X, y=y, groups=group)):
+        # Fill in indices with the training/test groups
+        indices = np.array([np.nan] * len(X))
+        indices[tt] = 1
+        indices[tr] = 0
+
+        # Visualize the results
+        ax.scatter(range(len(indices)), [ii + .5] * len(indices),
+                   c=indices, marker='_', lw=lw, cmap=cmap_cv,
+                   vmin=-.2, vmax=1.2)
+
+    # Plot the data labels and groups at the end
+    ax.scatter(range(len(X)), [ii + 1.5] * len(X),
+               c=y, marker='_', lw=lw, cmap=cmap_data)
+
+    ax.scatter(range(len(X)), [ii + 2.5] * len(X),
+               c=group, marker='_', lw=lw, cmap=cmap_data)
+
+    # Formatting
+    yticklabels = list(range(n_splits)) + ['labels', 'groups']
+    ax.set(yticks=np.arange(n_splits+2) + .5, yticklabels=yticklabels,
+           xlabel='Sample index', ylabel="CV iteration",
+           ylim=[n_splits+2.2, -.2], xlim=[0, 100],
+           title='{}'.format(type(cv).__name__))
+    return ax
+
+
+###############################################################################
+# Let's see how it looks for the `KFold` cross-validation object:
+
+fig, ax = plt.subplots()
+n_splits = 4
+cv = KFold(n_splits)
+plot_cv_indices(cv, X, y, groups, ax, n_splits)
+
+###############################################################################
+# As you can see, by default the KFold cross-validation iterator does not
+# take either datapoint label or group into consideration. We can change this
+# by using the ``StratifiedKFold`` like so.
+
+fig, ax = plt.subplots()
+cv = StratifiedKFold(n_splits)
+plot_cv_indices(cv, X, y, groups, ax, n_splits)
+
+###############################################################################
+# In this case, the cross-validation retained the same ratio of labels across
+# each CV split. Next we'll visualize this behavior for a number of CV
+# iterators.
+#
+# Visualize cross-validation indices for many CV objects
+# ------------------------------------------------------
+#
+# Let's visually compare the cross validation behavior for many
+# scikit-learn cross-validation objects. Below we will loop through several
+# common cross-validation objects, visualizing the behavior of each.
+#
+# Note how some use the group/label information while others do not.
+
+cvs = [KFold, GroupKFold, ShuffleSplit, StratifiedKFold,
```

Comment:
Do we need ``KFold`` and ``StratifiedKFold``?
Also, the plots seems a bit small from my side. See https://29038-843222-gh.circle-artifacts.com/0/doc/auto_examples/model_selection/plot_cv_indices.html

Reply:
re: the smallness of the plots, I'm not sure how I can control that since they're all independent figures - it looks like sphinx-gallery is styling them to fit in a grid pattern but something tells me I won't be able to control that behavior :-/

Reply:
So we can't do anything even with different figsize? 
Or maybe do not add legend for each subplot?